src/test/java/harness/utils/ExternalGCSBuckets.java

```diff
@@ -49,18 +50,35 @@ public static Bucket createBucket() throws IOException {
    * external project.
    */
   public static void grantReadAccess(Bucket bucket, String email) throws IOException {
+    grantAccess(
+        bucket,
+        Identity.user(email),
+        // TODO (PF-717): revisit this once we're calling WSM endpoints for check-access
+        StorageRoles.objectViewer(),
+        StorageRoles.legacyBucketReader());
+  }
+
+  /**
+   * Grant a given user admin access to a bucket. This method uses SA credentials for an external
+   * project.
```

Comment:
nit: Do we ever not want to use SA credentials for an external project? I'd think we could use the same methods whether or not the SA is in the same project as the bucket.

Reply:
Right, the SA doesn't need to be in the same project as the bucket. But the bucket needs to be in an external (to WSM) project in order for us to be able to set IAM policy on it. Practically, the SA will always be in the same project as the bucket, just because the CLI tests use a single project for testing referenced resources, but that's not required. Maybe this is just poor wording on my part, but if I've misunderstood your question, please restate.